### PR TITLE
Moved SwerveUtils to this branch

### PR DESCRIPTION
**Pull request by awfulbananas**
this vector class is great, we should use it anywhere applicable

### CODE REVIEW

File: src/main/java/frc/utils/Vector.java [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+// a class representing either a vector or a Point in space in 2 or more dimensions with double
+// precision
+// with operations for manipulating them
+
+// most operations affect the point they're called on, so for non-destructive operations, use
+// .copy()
+
+// when clockwise or counterclockwise is mentioned, keep in mind that in java, positive y is down,
+// not up
+public class Vector {
```
Comment (Irishumanoid) [inline src/main/java/frc/utils/Vector.java]: A bit misleading to use. Can you separate this class into a point class and a vector class?

File: src/main/java/frc/utils/Vector.java [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+  public Vector copy() {
+    Vector newP = new Vector(m_vals.length);
+    for (int i = 0; i < m_vals.length; i++) {
+      newP.set(i, m_vals[i]);
+    }
+    return newP;
+  }
+
+  // adds another dimension to this Point, initializing the value of that dimension to 0
+  public Vector addDim() {
```
Comment (Irishumanoid) [inline src/main/java/frc/utils/Vector.java]: when would we need to use this?

File: src/main/java/frc/utils/Vector.java [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+    for (int i = 0; i < dims() && i < min.dims(); i++) {
+      if (m_vals[i] < min.m_vals[i]) m_vals[i] = min.m_vals[i];
+      if (m_vals[i] > max.m_vals[i]) m_vals[i] = max.m_vals[i];
+    }
+    return this;
+  }
+
+  // transforms this Point into the space of Point space, then returns this Point
+  // this Point is also multiplied by the magnitude of space in the process,
+  // so if you want to avoid this, normalize space first
```
Comment (Irishumanoid) [inline src/main/java/frc/utils/Vector.java]: use case?

**Review by Irishumanoid — APPROVED** [review]

**Comment by awfulbananas** [comment]
I meant to actually merge this before re-adding the Vector class, but I didn't, so I force-pushed it back again (sorry)

**Pull request merged**